### DISCUSSION TITLE
Bad documentation

### DISCUSSION THREAD

**Discussion by darebgd** [cubiq/ComfyUI_IPAdapter_plus#390] [General]
I know writing documentation is the least favorite thing for a dev to do, but this is an example how poor docs make a product hard to use. 

Reading through requirements, I am utterly confused as to which one I need to download, whether I only need the first batch of Ip-adapters, and if so - why the second batch of ip-adapters (with faceid in the filename) have downloadable links - why do I need to download both batches and compare them, just to discover what I am supposed to do?

I mean - thx for the links, but do go a step forward and offer some more comprehensive way to explain what needs to be done with all the IP adapters (and there are many)

Not to mention two 'composition' models: I have no clue if I need to rename them or use them...

**Comment by cubiq** [comment]
this is what I can do with the given resources. which is more than many other developers are doing. do your part, make some research, it's not that hard
